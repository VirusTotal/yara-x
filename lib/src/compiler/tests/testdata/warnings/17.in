rule test {
  condition: 2 and 3
}