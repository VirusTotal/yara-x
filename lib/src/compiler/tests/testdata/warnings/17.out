warning: non-boolean expression used as boolean
 --> line:2:14
  |
2 |   condition: 2 and 3
  |              - this expression is `integer` but is being used as `bool`
  |
  = note: non-zero integers are considered `true`, while zero is `false`warning: non-boolean expression used as boolean
 --> line:2:20
  |
2 |   condition: 2 and 3
  |                    - this expression is `integer` but is being used as `bool`
  |
  = note: non-zero integers are considered `true`, while zero is `false`